docs/DeveloperGuide.md

```diff
@@ -65,10 +65,62 @@ When you are ready to start coding, we recommend that you look at the class diag
 code and the interaction among different classes.<br>
 
 ## Design & implementation
+### Reflection Component
+![Reflection Component Class Diagram](diagrams/ReflectionClassDiagram.png)
+This `Reflection` component provides users with random sets of introspective questions for users to reflect on.<br>
+<br>
+The `reflection` package consists of several classes, namely `ReflectionManager`, `SelfReflection`, `ReflectionQuestion`,
+`GetCommand`, `HomeCommand` and `ReflectUi`. There are three abstract classes `Manager`, `Command`, `TextUi` which some 
+classes in `reflection` package inherit from. But since these abstract classes are outside of `reflection` package, the 
+focus of this section will be on classes inside `reflection` package. <br>
+<br>
+
+`ReflectionManager` class:<br>
+- This class is in charge of the overall execution of the **Self Reflection** feature. 
+- It inherits from abstract class `Manager`
+- Each `ReflectionManager` object contains exactly one `ReflectUi` object as an attribute to get user inputs.
+- The `runEventDriver()` method is the entry of the Self Reflection feature, the caller creates a `ReflectionManager` 
+object and calls this method to launch the Self Reflection feature. It calls a class-level method `SelfReflection.geet()`
+to print greeting logo and message, therefore, `SelfRelfection` class is a dependency of `ReflectionManager` class. 
+- It contains a **while loop** to continuously get input commands from users until users input `home` command to return back to main WellNUS++ interface. 
+- The termination condition of the while loop is controlled by a static attribute `isExit`. Whenever `runEventDriver()`
+method is called, the `isExit` attribute will be initialised as `false`. This attribute can be accessed by other objects
+(more specifically `HomeCommand` object) through a static method `setIsExit()` to set to `true` and the while loop will 
+be terminated. 
+- Upon getting input commands, the `runEventDriver()` method will call the `executeCommands()` method. Based on the 
+input command type, the `executeCommands()` method will then create the correct type of command object and call `.execute()`
+method to execute the command accordingly.  
+
+`ReflectionQuestion` class:<br>
+- Each introspective question is a `ReflectionQuestion` object. 
+- It contains the basic description of the introspective question. 
+
+`SelfReflection` class:<br>
+- This class contains the information about the Self Reflection feature (e.g. greeting message, logo).
```

Comment:
Might be meaningful to explain _why_ SelfReflection is modelled as a `class` instead of just an `ArrayList<ReflectionQuestion>` directly in `ReflectionManager`, e.g. abstraction so `ReflectionManager` will only need to call `new SelfReflection()` and not be responsible for setting up the reflection questions, which is useful because reflection questions set up logic might be updated in the future